ansible_base/jwt_consumer/common/auth.py

```diff
@@ -120,10 +121,15 @@ def parse_jwt_token(self, request):
                 logger.warn(f"New user {self.user.username} created from JWT auth")
             except IntegrityError as exc:
                 logger.warning(f'Existing user {self.token["user_data"]} is a conflict with local user, error: {exc}')
-                self.user, created = get_user_model().objects.update_or_create(
-                    username=self.token["user_data"]['username'],
-                    defaults=user_defaults,
-                )
```

Comment:
This is one place where we didn't use `Resource.*_resource()` which created problems since we need a way to inject an internal attribute so we don't try to sync back to the resource server in this case. The below is my attempt to replicate this logic and also be able to inject the attribute before we `.save()`.

Reply:
This is an error case of dubious functionality to begin with. It might be better to just fail the authentication at this point. And you really changed the intent here, going from `update_or_create` to just `.get`. This code is reached when the "admin" user already exists, but with a different ansible_id. There are no good options in that case, but what this ends up doing looks worse than doing nothing.